### ISSUE TITLE
1.0.4 fails on publishing to github pages

### ISSUE THREAD

**Issue by chenglou** [facebook/Docusaurus#378]
### Is this a bug report?

Yes

### Have you read the [Contributing Guidelines](https://github.com/facebook/Docusaurus/blob/master/CONTRIBUTING.md)?

Yes

### Environment

CI (linux)

### Steps to Reproduce

Not too sure this repros on every project setup. See below

### Expected Behavior

Build succeeds.

### Actual Behavior

Error message:

```
Error: Copying build assets failed with error 'Error: Cannot copy '/home/ubuntu/reasonml.github.io/website/build' to a subdirectory of itself, '/home/ubuntu/reasonml.github.io/website/build/reasonml.github.io-master'.'
```

### Reproducible Demo

See the failure at https://circleci.com/gh/reasonml/reasonml.github.io/38, and the subsequent fix at https://circleci.com/gh/reasonml/reasonml.github.io/39 which is just a revert to docusaurus 1.0.3 (aka the old version used to work)

**Note**:

- My repro is a github org page. `projectName` is `""` in siteConfig.js. I recall such trick was necessary for a github org page, to publish to master. Not sure how relevant this detail is.
- Maybe this is related to the change from rsync to fs.copy?

cc @JoelMarcey, @rickyvetter

**Comment by JoelMarcey** [comment]
Thanks @chenglou -- yeah, I didn't expect that behavior to change from 1.0.3 to 1.0.4 -- so this is definitely a bug. Let me go through the relevant commits to see where we might have gone awry here.

**Comment by JoelMarcey** [comment]
@chenglou I have a test site that I am using to try to fix this...

https://github.com/JoelMarcey/JoelMarcey.github.io

I ran into the same bug you did, but only after I set both:

```
projectName: 'JoelMarcey.github.io',
organizationName: 'JoelMarcey',
```

in `siteConfig.js`.

This is because in order to test locally, these variables need to be set; If you leave those blank, it will use the the information from the CircleCI when running there as described here: https://docusaurus.io/docs/en/commands.html#docusaurus-publish

So I would recommend that you fill those out for clarity, but like you said, it is not strictly necessary if you are going to publish only through CircleCI.

So now on to figuring out why the bug is happening...

**Comment by JoelMarcey** [comment]
We use `fs.copy`, and this is probably the reason why it is failing:

https://github.com/jprichardson/node-fs-extra/issues/83

**Comment by JoelMarcey** [comment]
I think we got the fix in #384. I will push out a 1.0.5 bug fix release tomorrow/Wed so that these are part of the npm package.

**Comment by JoelMarcey** [comment]
1.0.5 has been pushed and this should be fixed now. Thanks for bringing this up.
